### ISSUE TITLE
observe()'s "old" option doesn't work for keypaths containing wildcards

### ISSUE THREAD

**Issue by supervacuo** [ractivejs/ractive#3118]
## Description:

The `old` option to `ractive.observe`, added in #2845, is ignored for keypaths containing wildcards

## Versions affected:

0.9.6, 1.0.0-build-83

## Platforms affected:

Tested on Firefox 56 on GNU/Linux

## Reproduction:

https://jsfiddle.net/1hrwbchp/5/

Basically, the following works:

```
ractive.observe("foo.bar", function(n, o, p) {
  console.log(o, n);
}, {
  old: function(prev, next) {
    if (next) {
      return next.slice();
    }
  }
});
```

but if you change the keypath to `*.bar`, the `old` function is never called, and `o` and `p` in the handler function are always the same as each other.

If this is expected behaviour (I notice the tests added in #2485 only check "plain" observers), any hints on how to have `o` be the previous value, as in Ractive 0.6.4, with wildcard observers?

**Comment by evs-chris** [comment]
I don't think I ever actually used the 0.6 series - went from 0.5.4 to 0.7 - but I didn't think it Ractive ever did any defensive copying. So if you pushed to an array watched by a pattern observer, the old value would be missing the new member?

At any rate, missing `old` for pattern observers was an oversight that has been corrected on edge, and will go out with 0.9.7 soon. Thanks for the report!

**Comment by supervacuo** [comment]
@evs-chris you're a hero, thank you <3

> I don't think I ever actually used the 0.6 series - went from 0.5.4 to 0.7 - but I didn't think it Ractive ever did any defensive copying. So if you pushed to an array watched by a pattern observer, the old value would be missing the new member?

No idea how or why it worked, but: https://jsfiddle.net/1hrwbchp/6/ (that's with 0.6.1)
